src/Macros/Recursive.php

```diff
@@ -17,7 +17,7 @@ public function __invoke()
     {
         return function (): Collection {
             return $this->map(function ($value) {
-                if (is_array($value) || is_object($value)) {
+                if (!($value instanceof \Closure) && (is_array($value) || is_object($value))) {
```

Comment:
This is unreadable to me. Could you split this up. Add a separate check to see if the value is a closure, and if so return `$value`.

Also import `Closure` here and in the test.